project_timeline/views/project_task_view.xml

```diff
@@ -58,11 +70,25 @@
             name="view_mode"
         >kanban,tree,form,calendar,timeline,pivot,graph,activity</field>
     </record>
+    <record id="project.action_view_all_task" model="ir.actions.act_window">
+        <field
+            name="view_mode"
+        >tree,kanban,form,calendar,timeline,pivot,graph,gantt,activity,map</field>
```

Comment:
On my instance I haven't `map` nor `gantt` views available, so I'm not able to see task...

So there are missing dependencies or we should remove those views here and define that in an other module. I would prefer to avoid adding those dependencies here and move those view modes somewhere else.